File: MobileApp/lib/models/user_profile.dart
import 'package:workout_app/models/past_schedule_day.dart';
import 'package:workout_app/models/workout_schedule.dart';

class UserProfile {
  int id;
  String userName;
  double? height;
  double? weight;
  double? maintenanceCalories;
  List<PastScheduleDay> pastScheduleDays;
  List<WorkoutSchedule> schedules;
  DateTime lastUpdated;

  UserProfile({
    required this.id,
    required this.userName,
    this.height,
    this.weight,
    this.maintenanceCalories,
    required this.pastScheduleDays,
    required this.schedules,
    required this.lastUpdated,
  });
}

extension UserProfileCopyWith on UserProfile {
  UserProfile copyWith({
    int? id,
    String? userName,
    double? height,
    double? weight,
    List<PastScheduleDay>? pastScheduleDays,
    List<WorkoutSchedule>? schedules,
    DateTime? lastUpdated,
  }) {
    return UserProfile(
      id: id ?? this.id,
      userName: userName ?? this.userName,
      height: height ?? this.height,
      weight: weight ?? this.weight,
      pastScheduleDays: pastScheduleDays ?? this.pastScheduleDays,
      schedules: schedules ?? this.schedules,
      lastUpdated: lastUpdated ?? this.lastUpdated,
    );
  }
}

File: API/WorkoutAPI/Models/UserProfile.cs
namespace WorkoutAPI.Models
{
    public class UserProfile
    {
        public required string UserName { get; set; }
        public double Height { get; set; } // Measured in inches
        public double Weight { get; set; } // Measured in pounds
        public WorkoutHistory History { get; set; }
        public List<WorkoutSchedule> Schedules { get; set; }
        public UserProfile() { }
        public UserProfile(string userName, double height, double weight, WorkoutHistory history, List<WorkoutSchedule> schedules)
        {
            UserName = userName;
            Height = height;
            Weight = weight;
            History = history;
            Schedules = schedules;
        }
    }
}
